lib/commands/generate.js

```diff
@@ -61,7 +53,17 @@ module.exports = Command.extend({
       this.project.initializeAddons();
     }
 
-    return task.run(taskOptions);
+    return this.runTask('GenerateFromBlueprint', taskOptions);
+  },
+
+  _env() {
+    return {
+      ui: this.ui,
+      analytics: this.analytics,
+      project: this.project,
+      testing: this.testing,
```

Comment:
Should we extend `Command._env()` with missing `settings` and `testing` props?

[Looks like](https://github.com/ember-cli/ember-cli/blob/ace2656c2637093d8443d907f70b45ae49b7ea30/lib/cli/cli.js#L103-L110) we always have these props inside commands.